File: day1/main.py
import itertools

with open('./input.txt') as f:
  data = map(int, f.read().strip().split('\n'))


for (a,b,c) in itertools.combinations(data, 3):
  if a+b+c == 2020:
    print (a*b*c)
    exit()

File: day1/main.hs
import Data.Set (insert, empty, member)
import Data.List (foldr, permutations)
solve:: [Integer] -> String
solve input = show $ head $ foldr (\x l -> if (2020-x) `member` s then (x*(2020-x)):l else l) [] input
  where s = foldr insert empty input

main :: IO()
main = interact ((\x -> x++['\n']) . solve . map read . lines)